go-controller/pkg/ovn/egressip.go

```diff
@@ -1496,11 +1510,11 @@ func (e *egressIPZoneController) addPodEgressIPAssignment(egressIPName string, s
 				return fmt.Errorf("unable to create NAT rule ops for status: %v, err: %v", status, err)
 			}
 		}
-		if config.OVNKubernetesFeature.EnableInterconnect && (loadedPodNode && !isLocalZonePod) {
+		if config.OVNKubernetesFeature.EnableInterconnect && !isOVNManagedNetwork && (loadedPodNode && !isLocalZonePod) {
 			// configure reroute for non-local-zone pods on egress nodes
-			ops, err = e.createStaticRouteOps(ops, podIPs, status, egressIPName, nextHopIP)
+			ops, err = e.createReroutePolicyOps(ops, podIPs, status, egressIPName, nextHopIP)
```

Comment:
@tssurya wouldn't this policy match traffic to a pod as well instead of just traffic egressing the cluster?

Reply:
hmm this LRP is added at 102, at 100 i think we have the default "allow" policies which match pod2pod traffic and ensure we don't send it out mp0, so this situation should be handled.
we should add more pod2pod + pod2external E2E's I have opened https://github.com/ovn-org/ovn-kubernetes/issues/3923

Reply:
```
13. lr_in_ip_routing (northd.c:10603): ip4.dst == 10.244.1.0/24, priority 74, uuid d6770c72
    ip.ttl--;
    reg8[0..15] = 0;
    reg0 = ip4.dst;
    reg1 = 10.244.1.1;
    eth.src = 0a:58:0a:f4:01:01;
    outport = "rtos-ovn-worker2";
    flags.loopback = 1;
    next;
14. lr_in_ip_routing_ecmp (northd.c:12285): reg8[0..15] == 0, priority 150, uuid 4541d25f
    next;
15. lr_in_policy (northd.c:9741): ip4.src == 10.244.0.0/16 && ip4.dst == 10.244.0.0/16, priority 102, uuid f50a1e6b
    reg8[0..15] = 0;
    next;
```
the 102 you see in the policy pipeline is what i mean